src/NLog/Targets/FileTarget.cs

```diff
@@ -2298,12 +2299,32 @@ private DateTime InitializeFile(string fileName, LogEventInfo logEvent)
             }
             else if (lastTime != now)
             {
+                now = EnsureValidLogEventTimeStamp(now, lastTime);
                 _initializedFiles[fileName] = now;
             }
 
             return lastTime;
         }
 
+        DateTime EnsureValidLogEventTimeStamp(DateTime logEventTimeStamp, DateTime previousTimeStamp)
+        {
+            if (logEventTimeStamp < previousTimeStamp && logEventTimeStamp.Date < previousTimeStamp.Date)
```

Comment:
when only comparing dates `if(logEventTimeStamp.Date < previousTimeStamp.Date)`, doesn't give that the same results

Reply:
Yes but `DateTime.Date` is a "slow" method. And by default then `logEventTimeStamp < previousTimeStamp` is never true. Thus skip calling the slow method.

Reply:
Ok please add a code comment for that, than I will merge it :)

Reply:
Added comment and added unit-test.